x-pack/plugins/infra/server/lib/alerting/log_threshold/log_threshold_executor.ts

```diff
@@ -66,28 +73,97 @@ export const createLogThresholdExecutor = (alertUUID: string, libs: InfraBackend
     }
   };
 
+interface ProcessingParams<Results> {
+  results: Results;
+  params: LogDocumentCountAlertParams;
+  alertInstanceFactory: AlertExecutorOptions['services']['alertInstanceFactory'];
+  alertId: string;
+}
+
+interface GroupBucket {
+  key: Record<string, string>;
+  doc_count: number;
+}
+
+interface ReducedGroupBucket {
+  name: string;
+  documentCount: number;
+}
+
+const processUngroupedResults = ({
+  results,
+  params,
+  alertInstanceFactory,
+  alertId,
+}: ProcessingParams<InfraDatabaseSearchResponse<{}, Aggregations>>) => {
+  const { count, criteria } = params;
+
+  const alertInstance = alertInstanceFactory(`${alertId}-${UNGROUPED_FACTORY_KEY}`);
+  const documentCount = results.hits.total.value;
```

Comment:
With the switch from `_count` to `hits` we would have to set `track_total_hits` to `true`, I think. Otherwise might only be accurate up to 10,000 hits by default according to the [ES docs](https://www.elastic.co/guide/en/elasticsearch/reference/7.7/search-request-body.html#request-body-search-track-total-hits):

> Generally the total hit count can’t be computed accurately without visiting all matches, which is costly for queries that match lots of documents. The track_total_hits parameter allows you to control how the total number of hits should be tracked. Given that it is often enough to have a lower bound of the number of hits, such as "there are at least 10000 hits", the default is set to 10,000. This means that requests will count the total hit accurately up to 10,000 hits. It’s is a good trade off to speed up searches if you don’t need the accurate number of hits after a certain threshold.

It seems, to optimize performance, we could also set the `track_total_hits` to the threshold we're checking.

This is, of course, only true for the ungrouped case. For the composite agg used in the grouped case it should already be accurate.